ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/query/LogsQuery.java

```diff
@@ -59,15 +59,7 @@ private boolean matchesTopics(final List<LogTopic> topics) {
   }
 
   private boolean matchesTopic(final LogTopic topic, final List<LogTopic> matchCriteria) {
-    for (final LogTopic candidate : matchCriteria) {
-      if (candidate == null) {
-        return true;
-      }
-      if (candidate.equals(topic)) {
-        return true;
-      }
-    }
-    return false;
+    return matchCriteria.isEmpty() || matchCriteria.contains(null) || matchCriteria.contains(topic);
```

Comment:
It looks like the previous behaviour would return false if the matchCriteria array was empty. And with this new logic, it will return true. Is this the desired behaviour?

Reply:
Huh, that doesn't sound like desired behavior. There were no tests testing the empty array case?

Reply:
I'm thinking an empty list should behave the same as a list containing "null".
It's not possible to get an empty list in the production system (due to how the json is parsed etc, but I think we need to get it right if we can ...